rpc/grpc/src/main/java/com/spotify/heroic/rpc/grpc/GrpcRpcClient.java

```diff
@@ -97,10 +98,11 @@ public void onMessage(final byte[] message) {
                 public void onClose(final Status status, final Metadata trailers) {
                     if (status.isOk()) {
                         if (!future.isDone()) {
-                            future.fail(new RuntimeException("Request finished without response"));
+                            future.fail(new RuntimeNodeException(getUri(),
```

Comment:
There are a few more cases where this might fail. Instead of mangling the exception here, it can be done on the cluster node layer to catch everything using `AsyncFuture#catchError(...)`.